snakemake/rules.py

```diff
@@ -276,9 +276,29 @@ def set_input(self, *input, **kwinput):
         consider_ancient = self.workflow.workflow_settings.consider_ancient.get(
             self.name, frozenset()
         )
-
         for i, item in enumerate(input):
-            self._set_inoutput_item(item, mark_ancient=i in consider_ancient)
+            if isinstance(item, collections.abc.Iterable) and not isinstance(item, str):
+                for ifile in item:
+                    if "multiext" in getattr(ifile, "_flags", {}):
+                        # Revert back multiext part to prefix, to avoid breaking cache later.
+                        _name = ifile._flags["multiext"].name
+                        ifile._flags["multiext"] = ifile._flags["multiext"].prefix
```

Comment:
Rather use get_flag_value(ifile, "multiext") here, saving the returned value into a variable and use that to do the work.